### PR TITLE
Update `read_*_if_necessary()` to return local arg name

### LINKED ISSUE #31: Add function to check and read input types
As suggested in review, we could add helper function to check and import spatRast and or vector inputs or filepaths. 

read_spatrast_if_necessary = function(spatRast){

if (inherits(spatRast, c("character"))) {
spatRast <- terra::rast(spatRast)

} else if (!inherits(spatRast, c("SpatRaster"))) {
cli::cli_abort("{.var spatRast} must be a SpatRaster or a path to a file")
}
return(spatRast)
}

`
Would need to update the functions where these are added in PEMprepr and PEMsamplr

### PR DESCRIPTION
**Pull request by ateucher**
This updates `read_spatrast_if_necessary()` and `read_sf_if_necessary()` so that when they error, they give the user the name of the argument, so they can be used in any context and still provide informative error messages. It doesn't require any changes where they're implemented, as it adds default argument values to the arg and call parameters, which are obtained from local context.

Also fixed some fragile SAGA-related testing infrastructure, mostly related to my computer where I have multiple SAGAs installed!

Closes #31

### CODE REVIEW

File: R/utils.R
Comment (gcperk) [inline R/utils.R:1]: This looks nifty. I was wondering how you would implement that, as I had no clue how to make it work!

File: tests/testthat/helper.R
Comment (gcperk) [inline tests/testthat/helper.R:1]: I also ran this function on my Linux machine and went very smoothly! (pre-pr).

**Review by gcperk — APPROVED** [review]
Great work - @ateucher that looks like a nice neat what to provide a much more specific input messages. Thanks for the improvements!

**Comment by ateucher** [comment]
thanks @gcperk !

**Pull request merged**